Domains/1-Common/DistributionSystems/DistributionSystems.ecschema.xml

```diff
@@ -62,4 +62,83 @@
             </IsMixin>
         </ECCustomAttributes>
     </ECEntityClass>
+
+    <ECEnumeration typeName="DirectionType" backingTypeName="int" isStrict="true">
+        <ECEnumerator value="0" displayLabel="Undefined" name="Undefined"/>
+        <ECEnumerator value="1" displayLabel="Input" name="Input"/>
+        <ECEnumerator value="2" displayLabel="Output" name="Output"/>
+        <ECEnumerator value="3" displayLabel="Input Or Output" name="InputOrOutput"/>
+        <ECEnumerator value="4" displayLabel="Input And Output" name="InputAndOutput"/>
+    </ECEnumeration>
+
+    <ECEntityClass typeName="DistributionPort" modifier="Abstract" displayLabel="Distribution Port" description="An inlet or outlet of a distribution element through which a particular substance may flow.">
+        <BaseClass>bis:SpatialLocationElement</BaseClass>
+        <ECProperty propertyName="Description" typeName="string" displayLabel="Description" description="A human-readable description of the Distribution Port"/>
+        <ECProperty propertyName="Direction" typeName="DirectionType" displayLabel="Direction" description="Direction of the port: e.g. Input, Output, Free Directional, Bi-Directional."/>
+        <ECProperty propertyName="LocalOrigin" typeName="point3d" description="The local placement origin of this DistributionPort in relation to its parent.">
+            <ECCustomAttributes>
+                <HiddenProperty xmlns="CoreCustomAttributes.01.00.03"/>
+            </ECCustomAttributes>
+        </ECProperty>
+        <ECProperty propertyName="LocalYaw" typeName="double" description="The local Yaw angle (in degrees) of the orientation of this DistributionPort in relation to its parent.">

```

Comment:
Isn't it against our guidelines to store values not in SI? Shouldn't all angles be stored in radians?

Reply:
can't recall why is this in angles, but I agree, we should also be able to set angle koq here.

Reply:
in the discussion, Diego pointed out the same properties in BisCore are also stored in degrees, so we decided to follow here.